exporters/otlp/test/otlp_log_recordable_test.cc

```diff
@@ -0,0 +1,237 @@
+  EXPECT_EQ(rec.log_record().span_id(), expected_bytes);
+}
+
+TEST(OtlpLogRecordable, SetResource)
+{
+  OtlpLogRecordable rec;
+  const std::string service_name_key = "service.name";
+  std::string service_name           = "test-otlp";
+  rec.SetResource(service_name_key, service_name);
+
+  auto proto_resource     = rec.ProtoResource();
+  bool found_service_name = false;
+  for (int i = 0; i < proto_resource.attributes_size(); i++)
+  {
+    auto attr = proto_resource.attributes(static_cast<int>(i));
+    if (attr.key() == service_name_key && attr.value().string_value() == service_name)
+    {
+      found_service_name = true;
+    }
+  }
+  EXPECT_TRUE(found_service_name);
+}
+
+TEST(OtlpLogRecordable, DefaultResource)
+{
+  OtlpLogRecordable rec;
+
+  auto proto_resource      = rec.ProtoResource();
+  int found_resource_count = 0;
+  for (int i = 0; i < proto_resource.attributes_size(); i++)
+  {
+    auto attr = proto_resource.attributes(static_cast<int>(i));
+    if (attr.key() ==
+        opentelemetry::sdk::resource::attr(OTEL_CPP_CONST_HASHCODE(AttrTelemetrySdkLanguage)))
+    {
+      EXPECT_EQ(attr.value().string_value(), "cpp");
+      ++found_resource_count;
+    }
+    else if (attr.key() ==
+             opentelemetry::sdk::resource::attr(OTEL_CPP_CONST_HASHCODE(AttrTelemetrySdkName)))
+    {
+      EXPECT_EQ(attr.value().string_value(), "opentelemetry");
+      ++found_resource_count;
+    }
+    else if (attr.key() ==
+             opentelemetry::sdk::resource::attr(OTEL_CPP_CONST_HASHCODE(AttrTelemetrySdkVersion)))
+    {
+      EXPECT_EQ(attr.value().string_value(), OPENTELEMETRY_SDK_VERSION);
+      ++found_resource_count;
+    }
+  }
+  EXPECT_EQ(3, found_resource_count);
+}
+
+// Test non-int single types. Int single types are tested using templates (see IntAttributeTest)
+TEST(OtlpLogRecordable, SetSingleAtrribute)
+{
+  OtlpLogRecordable rec;
+  nostd::string_view bool_key = "bool_attr";
+  common::AttributeValue bool_val(true);
+  rec.SetAttribute(bool_key, bool_val);
+
+  nostd::string_view double_key = "double_attr";
+  common::AttributeValue double_val(3.3);
+  rec.SetAttribute(double_key, double_val);
+
+  nostd::string_view str_key = "str_attr";
+  common::AttributeValue str_val(nostd::string_view("Test"));
+  rec.SetAttribute(str_key, str_val);
+
+  EXPECT_EQ(rec.log_record().attributes(0).key(), bool_key);
+  EXPECT_EQ(rec.log_record().attributes(0).value().bool_value(), nostd::get<bool>(bool_val));
+
+  EXPECT_EQ(rec.log_record().attributes(1).key(), double_key);
+  EXPECT_EQ(rec.log_record().attributes(1).value().double_value(), nostd::get<double>(double_val));
+
+  EXPECT_EQ(rec.log_record().attributes(2).key(), str_key);
+  EXPECT_EQ(rec.log_record().attributes(2).value().string_value(),
+            nostd::get<nostd::string_view>(str_val).data());
+}
+
+// Test non-int array types. Int array types are tested using templates (see IntAttributeTest)
+TEST(OtlpLogRecordable, SetArrayAtrribute)
```

Comment:
```suggestion
TEST(OtlpLogRecordable, SetArrayAttribute)
```

Reply:
Done. I also fix the spell error in `otlp_recordable_test.cc`